lib/chain_sync.ex

```diff
@@ -254,6 +254,130 @@ defmodule Xander.ChainSync do
     end
   end
 
+  # Handle socket messages that arrive during the race condition window
+  # between setting active mode and transitioning to new_blocks state
+  def catching_up(
```

Comment:
Nevermind the function body; it's just a copy of what's in `new_blocks/3`. My actual questions are:

1. Is this what the TODO is talking about; we need to handle incoming socket messages while in `catching_up` state?
2. Do you remember where how you saw this? I'd like to recreate it locally.